custom_components/sems/sensor.py

```diff
@@ -5,38 +5,567 @@
+                        f"{serial_number}-{idx}-bms_temperature",
+                        f"Inverter {inverter['name']} Battery {idx} BMS Temperature",
+                        path_to_battery + ["bms_temperature"],
+                        SensorDeviceClass.TEMPERATURE,
+                        TEMP_CELSIUS,
+                        SensorStateClass.MEASUREMENT,
+                    ),
+                    SensorOptions(
+                        device_info,
+                        f"{serial_number}-{idx}-bms_discharge_i_max",
+                        f"Inverter {inverter['name']} Battery {idx} BMS Discharge Max Current",
+                        path_to_battery + ["bms_discharge_i_max"],
+                        SensorDeviceClass.CURRENT,
+                        ELECTRIC_CURRENT_AMPERE,
+                        SensorStateClass.MEASUREMENT,
+                    ),
+                    SensorOptions(
+                        device_info,
+                        f"{serial_number}-{idx}-bms_charge_i_max",
+                        f"Inverter {inverter['name']} Battery {idx} BMS Charge Max Current",
+                        path_to_battery + ["bms_charge_i_max"],
+                        SensorDeviceClass.CURRENT,
+                        ELECTRIC_CURRENT_AMPERE,
+                        SensorStateClass.MEASUREMENT,
+                    ),
+                ]
+
+    if "hasPowerflow" in data and data["hasPowerflow"] and "powerflow" in data:
+        inverter_serial_number = get_home_kit_sn(data)
+        if not has_existing_homekit_entity or inverter_serial_number is None:
+            inverter_serial_number = "powerflow"
+        serial_backwards_compatibility = (
+            "homeKit"  # the old code uses homeKit for the serial number
+        )
+        device_info = DeviceInfo(
+            identifiers={
+                # Serial numbers are unique identifiers within a specific domain
+                (DOMAIN, serial_backwards_compatibility)
+            },
+            name="HomeKit",
+            manufacturer="GoodWe",
+        )
+        sensors += [
+            SensorOptions(
+                device_info,
+                f"{inverter_serial_number}",  # backwards compatibility otherwise would be f"{serial_number}-load"
+                f"HomeKit Load",
+                ["powerflow", "load"],
+                SensorDeviceClass.POWER,
+                POWER_WATT,
+                SensorStateClass.MEASUREMENT,
+            ),
+            SensorOptions(
+                device_info,
+                f"{inverter_serial_number}-pv",
+                f"HomeKit PV",
+                ["powerflow", "pv"],
+                SensorDeviceClass.POWER,
+                POWER_WATT,
+                SensorStateClass.MEASUREMENT,
+            ),
+            SensorOptions(
+                device_info,
+                f"{inverter_serial_number}-grid",
+                f"HomeKit Grid",
+                ["powerflow", "grid"],
```

Comment:
gridStatus should be used to determine buy/sell.

Reply:
Do you know what the possible values of `gridStatus` are and what they mean?

Reply:
gridStatus seems to not work, it is always 1 for me. I have pmeter in the inverter device though and that shows same value as grid from powerflow but with sign. pmeter is negative when buying and positive when selling. Note that I don't have homekit smart meter, I have GM3000 smart meter.